sdks/java/core/src/main/java/org/apache/beam/sdk/util/NumberedShardedFile.java

```diff
@@ -188,14 +193,16 @@ public String toString() {
    * of given files equals the number that is parsed from shard name.
    */
   @VisibleForTesting
-  boolean checkTotalNumOfFiles(Collection<String> files) {
-    for (String filePath : files) {
-      Path fileName = Paths.get(filePath).getFileName();
+  boolean checkTotalNumOfFiles(Collection<Metadata> files) {
+    for (Metadata fileMedadata : files) {
+      String fileName = fileMedadata.resourceId().toString().substring(
```

Comment:
No, but I think we probably need it. It would actually make a lot of textio stuff much nicer.

Add it!